File: corec.r



rjson_exists <- require('rjson')

if (! rjson_exists) {
	print ('rjson does not exist. Installing it..');
	install.packages("rjson", repos='http://cran.us.r-project.org');
}

library(rjson);

corec_get <- function(parameter) {
	json_data <- fromJSON(file="corec_parameters.json");
	return(json_data[[parameter]]);
}

corec_set <- function(parameter, value) {
	json_data <- fromJSON(file="corec_parameters.json");
	json_data[parameter] <- value;

	json_str <- toJSON(json_data);
	write(json_str, 'corec_parameters.json');
}

corec_lock <- function(lock) {
	json_data <- fromJSON(file="corec_locks.json");
	json_data[lock] <- TRUE;

	json_str <- toJSON(json_data);
	write(json_str, 'corec_parameters.json');
}

corec_unlock <- function(lock) {
	json_data <- fromJSON(file="corec_locks.json");
	json_data[lock] <- FALSE;

	json_str <- toJSON(json_data);
	write(json_str, 'corec_parameters.json');
}

File: corec.py

import json

def corec_set(parameter, value):

	params_fn = "corec_parameters.json"

	with open(params_fn) as f:
		params = json.load(f)

	params[parameter] = value

	with open(params_fn, 'w') as f:
		json.dump(params, f, indent=4)

def corec_get(parameter):

	params_fn = "corec_parameters.json"

	with open(params_fn) as f:
		params = json.load(f)

	if parameter in params:
		return params[parameter]

	return None

def corec_lock(lock):
	locks_fn = "corec_locks.json"

	with open(locks_fn) as f:
		locks = json.load(f)

	locks[lock] = True

	with open(locks_fn, 'w') as f:
		json.dump(locks, f, indent=4)

def corec_unlock(lock):
        locks_fn = "corec_locks.json"

        with open(locks_fn) as f:
                locks = json.load(f)

        locks[lock] = False

        with open(locks_fn, 'w') as f:
                json.dump(locks, f, indent=4)



